src/ensembl/src/content/app/entity-viewer/gene-view/components/default-transcripts-list/transcripts-list-item-info/TranscriptsListItemInfo.tsx

```diff
@@ -123,6 +128,20 @@ export const TranscriptsListItemInfo = (
     objectId: props.gene.id
   });
 
+  const showProtein = () => {
+    const transcriptId = transcript.id;
+
+    props.expandProtein(transcriptId);
+
+    const url = urlFor.entityViewer({
+      genomeId,
+      entityId,
+      view: View.PROTEIN
+    });
+
+    props.push(url);
+  };
```

Comment:
Looks like we could just use the `Link` component?

Reply:
>Regarding the expandProtein part, do we possibly want to add the focus protein to the url?

I thought the same but I am not sure when we need to update the focus protein. Do we do it on mouseover?  
Also, if we don't update it and if the user revisits the proteins list, the focus will always be the same.

Reply:
We may set just a unidirectional logic:

- if there is a focus-protein parameter in the url when the view is protein,
- then expand corresponding protein (if it exists)
- and scroll to the expanded protein

Not sure what should happen when user clicks on another protein. They should be allowed to expand multiple proteins.

Hmm... Maybe having focus protein id in the url is not such a good idea after all.

Reply:
I did think about this approach of having the focus parameter in the urls but I was wondering if we could simply scroll to the id that was most recently added to `expandedTranscriptIds`. With this approach, when the user clicks on a protein id, it will get added to the `expandedTranscriptIds` and will become the most recently added id. We will of course need to handle clicking on an already expanded id in which case we will have to pop it and add to the end.

Reply:
> if we could simply scroll to the id that was most recently added to expandedTranscriptIds

There are potential problems with this approach:
- I am not sure we even need to apply programmatic scrolling when the user is already on the protein page and is clicking on proteins to expand or collapse them. In this case, the user already sees the protein he is interacting with.
- How are we planning to handle page refreshes? Also, do we plan to be able to share the url with a particular expanded protein, or do we give up on this idea?